code/obj/machinery/runetrap.dm

```diff
@@ -0,0 +1,228 @@
+				playsound(src, "sound/effects/electric_shock_short.ogg", 30, 1)
+				src.visible_message("<span class='alert>[M] steps on [src] and triggers it! Your hair stands on end!</span>")
+				playsound(src, "sound/voice/wraith/wraithraise3.ogg", 80)
+				elecflash(src, 1, 1)
+				qdel(src)
+
+/obj/machinery/wraith/runetrap/terror
+	Crossed(atom/movable/A)
+		..()
+		if(!src.armed)
+			return 1
+		if(ismob(A) && !istype(A, /mob/living/critter/trickster_puppet) && !istype(A, /mob/wraith))
+			var/mob/M = A
+			if(checkRun(M))
+				for (var/mob/H in range(4, src))
+					H.setStatus("terror", 45 SECONDS)
+
+				src.visible_message("<span class='alert>[M] steps on [src] and triggers it! Your mind fills with terrible visions!</span>")
+				playsound(src, "sound/voice/wraith/wraithraise3.ogg", 80)
+				elecflash(src, 1, 1)
+				qdel(src)
+
+/obj/machinery/wraith/runetrap/fire
+	Crossed(atom/movable/A)
+		..()
+		if(!src.armed)
+			return 1
+		if(ismob(A) && !istype(A, /mob/living/critter/trickster_puppet) && !istype(A, /mob/wraith))
+			var/mob/M = A
+			if(checkRun(M))
+				fireflash(M, 1, TRUE)
+				playsound(src, "sound/effects/mag_fireballlaunch.ogg", 50, 0)
+				src.visible_message("<span class='alert>[M] steps on [src] and triggers it! A flame engulfs them immediatly!</span>")
+				playsound(src, "sound/voice/wraith/wraithraise3.ogg", 80)
+				elecflash(src, 1, 1)
+				qdel(src)
+
+/obj/machinery/wraith/runetrap/teleport
+	Crossed(atom/movable/A)
+		..()
+		if(!src.armed)
+			return 1
+		if(ismob(A) && !istype(A, /mob/living/critter/trickster_puppet) && !istype(A, /mob/wraith))
+			var/mob/M = A
+			if(checkRun(M))
+				var/telerange = 5
+				var/list/randomturfs = new/list()
+				if (isrestrictedz(M.z))
+					src.visible_message("<span class='alert>[M] steps on [src] and triggers it! It missfires, sputters and dies!</span>")
+					elecflash(src, 2, 1)
+					qdel(src)
+					return 1
+				for(var/turf/T in orange(M, telerange))
+					if(istype(T, /turf/space) || T.density) continue
+					randomturfs.Add(T)
+				if (length(randomturfs) <= 0)
+					return 1
+				src.visible_message("<span class='alert>[M] steps on [src] and triggers it! A flame engulfs them immediatly!</span>")
+				playsound(src, "sound/voice/wraith/wraithraise3.ogg", 80)
+				SPAWN(5 DECI SECONDS)
+					boutput(M, text("<span class='alert'>You blink, and suddenly you're somewhere else!</span>"))
+					playsound(M.loc, "sound/effects/mag_warp.ogg", 25, 1, -1)
+					M.set_loc(pick(randomturfs))
+					elecflash(src, 1, 1)
+					qdel(src)
+
+/obj/machinery/wraith/runetrap/explosive
+	Crossed(atom/movable/A)
+		..()
+		if(!src.armed)
+			return 1
+		if(ismob(A) && !istype(A, /mob/living/critter/trickster_puppet) && !istype(A, /mob/wraith))
+			var/mob/M = A
+			if(checkRun(M))
+				src.visible_message("<span class='alert>[M] steps on [src] and triggers it! You hear a buzzing sound!</span>")
+				playsound(src, "sound/voice/wraith/wraithraise3.ogg", 80)
+				SPAWN(5 DECI SECOND)
```

Comment:
why

Reply:
Was initially meant as a way to make explosions a bit less dangerous since it lets people walk a tile or two away before it blows up. I suppose it's not needed, removing it.